### ISSUE TITLE
Variable microbial stoichiometry

### ISSUE THREAD

**Issue by wwieder** [wwieder/biogeochem_testbed_1.1#41]
The first implementation of MIMICS-cn uses fixed stoichoimetry for MICr and MICK (6 & 10, respectively), resulting in a weak / absent latitudinal gradient that is inconsistent with observations from Xu et al. [2013](https://onlinelibrary.wiley.com/doi/abs/10.1111/geb.12029). 

Alternative strategies would be to have microbial C:N be a function of litter quality (see Liu et al. [2019](https://link-springer-com.cuucar.idm.oclc.org/article/10.1007/s00374-019-01423-1), inorganic N availability (as in C:N pools w/in DAYCENT), or others?

**Comment by wwieder** [comment]
@melanniehartman I tried adding this feature with these code modifications.
the model seems to be running, do you mind taking a look at the code differences here?
https://github.com/wwieder/biogeochem_testbed_1.1/commit/e4ea7f3a100809e4a1e4377b8de69e398da5377e

**Comment by melanniehartman** [comment]
Hi Will,

Your updates look good to me! I was just wondering about the modification
to microbial C:N that is dependent on litter.

*mimics_cycle_CN.f90 lines 1595-1595:*

So the microbe C:N increases with the metabolic litter fraction?  I had
been thinking it would increase when microbes consume more structural (high
C:N) litter (when fmet decreases).

I see you added a new parameter, mimicsbiome%cnModDenom.  What kind of
value does it have?



mimicsbiome%CN_r(npt) = mimicsbiome%CNr *
SQRT(mimicsbiome%fmet(npt)/mimicsbiome%cnModDenom)

      mimicsbiome%CN_k(npt) = mimicsbiome%CNk *
SQRT(mimicsbiome%fmet(npt)/mimicsbiome%cnModDenom)

**Comment by wwieder** [comment]
oh, you're @melanniehartman right I did this the wrong way, leading to increases in micCN with higher quality letter, which was not intended. See fig on the left (with variable micCN, vs. the baseline case)
![image](https://user-images.githubusercontent.com/8031012/99730748-321bee00-2a7a-11eb-8f8c-b4850718a382.png)

**Comment by melanniehartman** [comment]
I imagine you'll just use 1-fmet instead? Looks like the modification makes
a big difference. It will be interesting to see the larger impact.

**Comment by wwieder** [comment]
Yep, I just flipped the function result on the left are with the modified C:N, the default is on right.
![image](https://user-images.githubusercontent.com/8031012/99735942-c1c59a80-2a82-11eb-9e7d-338345535ee5.png)

**Comment by wwieder** [comment]
target microbial C:N switches too quickly, causing crazy behavior.  @melanniehartman what would cause fMET to change so much at a single time?  These results are from a single grid cell exhibiting this behavior of the 20th century.

![image](https://user-images.githubusercontent.com/8031012/99996715-5d078a00-2d79-11eb-847a-76677e7b1efb.png)

**Comment by wwieder** [comment]
I think this has something to do with increases slight changes in inorganic N that increase wood decay, resulting in big shifts fMET, which feedback to further increase N mineralization rates.  I've tried turning off N limitation of wood decay in casa_inout for MIMICS to test this.  then we can discuss...

**Comment by wwieder** [comment]
This whole section of code could be removed if we want wood decay to be unaffected by N availability:
https://github.com/wwieder/biogeochem_testbed_1.1/blob/165716f0875875767f11c0a27aafae693d5bc4be/SOURCE_CODE/casa_inout.f90#L1610

**Comment by wwieder** [comment]
Also needed to but min value on Lignin:N ratios to keep fmet from being too low (or negative) in forests with higher CWDc flux.  
- Currently capped at 40 and ~0.25, respectively.

**Comment by wwieder** [comment]
Wood dynamics look better with revisions mentioned above and a smoother accumulation of steadystate C pools (here for the last 10k years of a 20k spinup.

![image](https://user-images.githubusercontent.com/8031012/100639727-c98f0580-32f2-11eb-9514-383431583a71.png)
